### PR TITLE
Emscripten compiler support

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2/Context.cpp [hunk 1]
```diff
@@ -104,6 +107,9 @@ namespace OpenRCT2
         // Remove this when GetContext() is no longer called so that
         // multiple instances can be created in parallel
         static Context * Instance;
+        static void RunFixedFrameFunc() {
+            Instance->RunFixedFrame();
+        }
```
Comment (IntelOrca) [inline src/openrct2/Context.cpp]: I thought we were trying to discourage singletons?

Comment (wolfreak99) [reply src/openrct2/Context.cpp]: I can see this being only a temporary thing, with the way the comment words it. Sometimes you have to start small, especially in a script system, (that is, if this is part of the script / plugin support), you want to make sure it at least functions, and then work your way up to not using singletons.

**Comment by janisozaur** [comment]
@IntelOrca updated

**Comment by janisozaur** [comment]
@IntelOrca is this ready then?

**Comment by IntelOrca** [comment]
Yep

**Comment by rjy7wb** [comment]
any guide for actually targeting webassembly?

**Comment by danilw** [comment]
emscripten 1.38.31 crashes with segfault(mayne other projects work with no errors there)
clang++: error: unable to execute command: Segmentation fault (core dumped)

when i tried build this project OpenRCT2 (I have setup all library and everything else), crashes on first file to compile that is CursorRepository.cpp

maybe someone can make propper bugreport for clang(emscripten)

**Comment by janisozaur** [comment]
Since you already have the setup, can you do that?

**Comment by danilw** [comment]
> can you do that?
generated file for bugreport is 69000 lines long(no one gona read it)...and I not sure if it not my mistake somewhere, so nwm sorry
maybe it really my mistake somewhere(but I think compiler should not crash on file compiling...when other projects work okey)

**Comment by greatjack1** [comment]
Currently getting this error when trying to compile using the latest emscripten.

Make Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR) (Required is at least version "1.0.0")
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.16/Modules/FindOpenSSL.cmake:447 (find_package_handle_standard_args)
  src/openrct2/CMakeLists.txt:45 (find_package)
  CMakeLists.txt:363 (include)


Any advice?

System is ubuntu 20.01 and openssl is installed.

**Comment by janisozaur** [comment]
You need to disable (at least) network and http in build config

**Comment by janisozaur** [comment]
spry0P3

**Pull request merged**